azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/JoinCallButtonHolderView.kt

```diff
@@ -83,4 +172,41 @@ internal class JoinCallButtonHolderView : ConstraintLayout {
             joiningCallText.visibility = GONE
         }
     }
+
+    private fun isNetworkConnectionAvailable(): Boolean {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            val capabilities = connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+
+            if (capabilities != null) {
+                return capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) ||
+                    capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ||
+                    capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)
+            }
+        } else {
+            val networkInfo = connectivityManager.activeNetworkInfo
+            networkInfo ?: return false
+            return networkInfo.isConnected && (
+                networkInfo.type == ConnectivityManager.TYPE_WIFI ||
+                    networkInfo.type == ConnectivityManager.TYPE_MOBILE
+                ) ||
+                networkInfo.type == ConnectivityManager.TYPE_ETHERNET
+        }
+
+        return false
+    }
```

Comment:
Can we have this code separate in a utility class so that it might become a reusable stub of code?